docs/PumpChecklist.md

```diff
@@ -0,0 +1,213 @@
+
+### Required if Present
+
+#### Basals
+
+  - `[ ]` scheduled basal
+    - `[ ]` basal rate intervals with a start time, duration, and rate delivered
+    - `[ ]` name of basal schedule on each scheduled basal rate interval
+  - `[ ]` manual temp basal
+    - `[ ]` basal rate intervals with a start time, duration, and rate delivered
+    - `[ ]` object representing suppressed scheduled basal *for each segment of the basal schedule that the temp basal intersects*
+  - `[ ]` percentage temp basal
+    - `[ ]` basal rate intervals with a start time, duration, percent
+        - `[ ]` rate provided directly OR
+        - `[ ]` rate computed from percent x suppressed.rate
+    - `[ ]` object representing suppressed scheduled basal *for each segment of the basal schedule that the temp basal intersects*
+  - `[ ]` "suspended" basals (see [status - suspends & resumes)[#device-events] below)
+    - `[ ]` basal interval with a start time and duration but no rate (b/c suspended)
+    - `[ ]` object representing suppressed scheduled basal *for each segment of the basal schedule that the suspension of insulin delivery intersects*
+  - `[ ]` final (most recent) basal
+    - `[ ]` basal rate interval with a start time, duration "guessed" from settings, rate delivered, and an annotation re: the "guessed" duration OR
+    - `[ ]` basal rate interval with a start time and rate, no (= zero) duration
+
+Device-specific? (Add any device-specific notes/additions here.)
+
+#### Boluses
+
+  - `[ ]` normal bolus
+    - `[ ]` amount of insulin delivered
+    - `[ ]` amount of insulin delivery programmed (if differs from actual delivery, in case of bolus interruption, cancellation, etc.)
+  - `[ ]` extended bolus
+    - `[ ]` amount of insulin delivered
+    - `[ ]` duration of insulin delivery
+    - `[ ]` amount of insulin delivery programmed (if differs from actual delivery, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` duration of insulin delivery programmed (if differs from actual duration, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` extended bolus that crosses midnight is split into two records
+  - `[ ]` combo/dual bolus
+    - `[ ]` amount of insulin delivered - immediate (normal)
+    - `[ ]` amount of insulin delivered - extended
+    - `[ ]` duration of extended insulin delivery
+    - `[ ]` amount of immediate insulin delivery programmed (if differs from actual delivery, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` amount of extended insulin delivery programmed (if differs from actual delivery, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` duration of extended insulin delivery programmed (if differs from actual duration, in case of bolus interruption, cancellation, etc.)
+    - `[ ]` extended portion of combo bolus that crosses midnight is split into two records
+  - bolus cancellations/interruptions
```

Comment:
I believe the places where I've left off the checkbox next to something (and only included checkboxes in sub-items) make sense. If they don't make sense to you, LMK.